21-inferential-analysis.Rmd

```diff
@@ -1,6 +1,353 @@
+
+$$
+\log(\lambda) = \beta_0 + \beta_1x_1 + \ldots + \beta_px_p
+$$ 
+
+where $\lambda$ is the expected value of the counts.
+
+Let's fit a simple model that contains all of the predictor columns. The `r pkg(poissonreg)` package will create this model specification:
+
+```{r inferential-glm}
+library(poissonreg)
+
+# default engine is 'glm'
+log_lin_spec <- poisson_reg()
+
+log_lin_fit <- 
+  log_lin_spec %>% 
+  fit(art ~ ., data = bioChemists)
+log_lin_fit
+```
+
+The `tidy()` method succinctly summarizes the coefficients for the model (along with 90% confidence intervals).
+
+```{r inferential-glm-tidy}
+tidy(log_lin_fit, conf.int = TRUE, conf.level = 0.90)
+```
+
+Looking at these results, `phd` (the prestige of their department) may not have any relationship with the outcome.
+
+While the Poisson distribution is the routine assumption for data like these, it may be beneficial to conduct a rough check the model assumptions but fitting the models without using the Poisson likelihood to calculate the confidence intervals. The `r pkg(rsample)` package has a convenience function to compute bootstrap confidence intervals for `lm()` and `glm()` models. We can use this function, while explicitly declaring `family = poisson`, to compute a large number of model fits. By default, we compute a 90% confidence bootstrap-t interval but percentile and BCa intervals are also available:
+
+```{r inferential-glm-ci}
+set.seed(2103)
+glm_boot <- 
+  reg_intervals(art ~ ., data = bioChemists, model_fn = "glm", family = poisson)
+glm_boot
+```
+
+When we compare these results to the purely parametric results from `glm()`, the bootstrap intervals are somewhat wider. If the data were truly Poisson, these intervals would have more similar widths.
+
+```{r inferential-glm-intervals, echo = FALSE}
+glm_boot %>% 
+  select(term, method = .method, .estimate, .lower, .upper) %>% 
+  bind_rows(
+    tidy(log_lin_fit, conf.int = TRUE, conf.level = 0.90) %>% 
+      filter(term != "(Intercept)") %>% 
+      mutate(method = "parametric") %>% 
+      select(term, method, .estimate = estimate, .lower = conf.low, .upper = conf.high)
+  )%>%
+  ggplot(aes(x = .estimate, y = term, col = method)) +
+  geom_vline(xintercept = 0, lty = 3) +
+  geom_point(position = position_dodge(width = 1 / 2), cex = .9) +
+  geom_errorbar(aes(xmin = .lower, xmax = .upper),
+                width = 1 / 4,
+                position = position_dodge(width = 1 / 2)) +
+  labs(x = "GLM coefficients", y = NULL)
+```
+
+Determining which predictors to include in the model is a difficult problem. One approach to do conduct likelihood ratio tests [@McCullaghNelder89] between nested models. Based on the confidence intervals, there is evidence that a simpler model without `phd` may be sufficient. Let's fit a smaller model, the conduct a statistical test, and tidy the results:
```

Comment:
Also, since it says "conduct a statistical test" it's useful to outline what the hypotheses of that test are. (This is relevant here and also further below where `anova()` is used.